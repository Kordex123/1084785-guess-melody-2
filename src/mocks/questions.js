const questions = [
  {
    type: `genre`,
    genre: `pop`,
    answers: [
      {
        src: `https://youtu.be/n4RjJKxsamQ?t=69`,
        genre: `rock`,
      },
      {
        src: `https://youtu.be/6M6samPEMpM?t=160`,
        genre: `pop`,
      },
      {
        src: `https://www.youtube.com/watch?v=TTA2buWlNyM`,
        genre: `country`,
      },
      {
        src: `https://youtu.be/p47fEXGabaY?t=28`,
        genre: `pop`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `folk`,
    answers: [
      {
        src: `https://www.youtube.com/watch?v=TTA2buWlNyM`,
        genre: `country`,
      },
      {
        src: `https://www.youtube.com/watch?v=OUeaAOIAbXs`,
        genre: `hip-hop`,
      },
      {
        src: `https://www.youtube.com/watch?v=lRprww9jrU0`,
        genre: `folk`,
      },
      {
        src: `https://youtu.be/p47fEXGabaY?t=28`,
        genre: `pop`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Cristina Aguilera`,
      src: `https://youtu.be/-ScjucUV8v0?t=48`,
    },
    answers: [
      {
        picture: `https://s1.r29static.com/bin/entry/5cc/340x408,85/2166451/image.webp`,
        artist: `Katy Perry`,
      },
      {
        picture: `https://www.rollingstone.com/wp-content/uploads/2019/03/shutterstock_10112916nc.jpg?resize=900,600&w=1200`,
        artist: `Lady Gaga`,
      },
      {
        picture: `https://s2.r29static.com/bin/entry/388/340x408,85/2225382/image.webp`,
        artist: `Cristina Aguilera`,
      },
    ],
  },
];

export default questions;
