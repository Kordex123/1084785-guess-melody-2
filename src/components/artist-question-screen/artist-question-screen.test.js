import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from "./artist-question-screen";
import questions from "../../mocks/questions";

it(`ArtistQuestionScreen correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<ArtistQuestionScreen
      screenIndex={2}
      question={questions[2]}
      onAnswer={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
