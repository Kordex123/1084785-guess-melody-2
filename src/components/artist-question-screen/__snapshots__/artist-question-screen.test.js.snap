// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly renders after relaunch 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-20"
          name="answer"
          type="radio"
          value="answer-20"
        />
        <label
          className="artist__name"
          htmlFor="answer-20"
        >
          <img
            alt="Katy Perry"
            className="artist__picture"
            src="https://s1.r29static.com/bin/entry/5cc/340x408,85/2166451/image.webp"
          />
          Katy Perry
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-21"
          name="answer"
          type="radio"
          value="answer-21"
        />
        <label
          className="artist__name"
          htmlFor="answer-21"
        >
          <img
            alt="Lady Gaga"
            className="artist__picture"
            src="https://www.rollingstone.com/wp-content/uploads/2019/03/shutterstock_10112916nc.jpg?resize=900,600&w=1200"
          />
          Lady Gaga
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-22"
          name="answer"
          type="radio"
          value="answer-22"
        />
        <label
          className="artist__name"
          htmlFor="answer-22"
        >
          <img
            alt="Cristina Aguilera"
            className="artist__picture"
            src="https://s2.r29static.com/bin/entry/388/340x408,85/2225382/image.webp"
          />
          Cristina Aguilera
        </label>
      </div>
    </form>
  </section>
</section>
`;
